src/vm/typehashingalgorithms.h

```diff
@@ -0,0 +1,90 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+// ---------------------------------------------------------------------------
+// Generic functions to compute the hashcode value of types
+// ---------------------------------------------------------------------------
+
+#pragma once
+#include <stdlib.h>
+
+// DIFFERENT FROM CORERT: Hashing code currently different from CoreRT
+
+//
+// Returns the hashcode value of the 'src' string
+//
+inline static DWORD ComputeNameHashCode(LPCUTF8 src)
+{
+    if (src == NULL || *src == '\0')
+        return 0;
+
+    DWORD hash1 = 0x6DA3B944;
+    DWORD hash2 = 0;
+
+    for (COUNT_T i = 0; src[i] != '\0'; i += 2)
+    {
+        hash1 = (hash1 + _rotl(hash1, 5)) ^ src[i];
+        if (src[i + 1] != '\0')
+            hash2 = (hash2 + _rotl(hash2, 5)) ^ src[i + 1];
+        else
+            break;
+    }
+
+    hash1 += _rotl(hash1, 8);
+    hash2 += _rotl(hash2, 8);
+
+    return hash1 ^ hash2;
+}
+
+inline static DWORD ComputeNameHashCode(LPCUTF8 pszNamespace, LPCUTF8 pszName)
+{
+    return ComputeNameHashCode(pszNamespace) ^ ComputeNameHashCode(pszName);
+}
+
+inline static DWORD ComputeArrayTypeHashCode(DWORD elementTypeHashcode, DWORD rank)
+{
+    // Arrays are treated as generic types in some parts of our system. The array hashcodes are 
```

Comment:
This comment is not true because of the algorithm is different from CoreRT. Delete it and just add comment that say that it is different from CoreRT.

I would change the whole implementation to be just: `return 0xd5313556 + rank;` (ie return same hashcode for single dimensional arrays as CoreRT, but different one for multidimensional arrays).